### ISSUE TITLE
Not working on firefox and chrome for wdio

### ISSUE THREAD

**Issue by RollerIn** [bewisse/modheader_selenium#24]
![image](https://user-images.githubusercontent.com/17290705/154622351-c17d1ffb-7906-4860-ac3f-b6aa59e70f5c.png)

I have tested this on chrome browser for WebdriverIO but did not work

even when I try to send more than one value that also considering as object

one more doubt onPrepare hook , we are creating 'browser.url(chromeModheader.getAddHeaderUrl('accept-encoding', ''));'
but in case of I want to test chrome and firefox at a time <parallelrun> how can we make this ? can we make as below right?
browser.url(chromeModheader.getAddHeaderUrl('accept-encoding', ''));'
browser.url(firefoxModheader.getAddHeaderUrl('accept-encoding', ''));'
will this header accept 'x-ak-clientip' and 'x-forwarded-for'?

**Comment by hao1300** [comment]
Can you provide the code snippet that does not work?

**Comment by RollerIn** [comment]
Config:
```
services: ['selenium-standalone','firefox-profile'],
        seleniumLogs: './logs',
        firefoxProfile: {
            extensions: [firefoxModheader.getExtension()],
            'xpinstall.signatures.required': false,
            legacy: false,
        },
        
capabilities: [
    {
        maxInstances: 1,
        browserName: 'chrome',
        'goog:chromeOptions': {
            extensions: [chromeModheader.getExtension()],
        },
    },
```
onPrepare:
```
let headerList=[];
var headers = {};
headers['x-ak-clientip'] = '100.145.10.11';
headers['x-forwarded-for'] = '100.145.10.11';
headerList.push(headers);
 //driver.get("https://webdriver.bewisse.com/add?" + headerList);
 const modHeader=chromeModheader.getAddHeaderUrl(headerList )
 browser.url( modHeader)
```
this is for chrome 

for firefox:
am not sure how to proceed on on prepare hook?

am I doing anything wrong here?

**Comment by RollerIn** [comment]
one more u[date, I took new new library  , now we can send multi params but error page is loading
https://webdriver.modheader.com/add?x-ak-clientip=100.145.10.11&x-forwarded-for=100.145.10.11---> error page
need additional info how can I navigate to my url ...browser.url("merl")...did not work

**Comment by hao1300** [comment]
So it appears that your configuration is incorrect. I have added an example for Webdriver IO here: https://github.com/modheader/modheader_selenium/blob/main/examples/webdriverio/wdio.conf.js. Please take a look.

Essentially, the `firefoxProfile` need to be passed in alongside `firefox-profile` service.

**Comment by RollerIn** [comment]
Yes, Essentially, the firefoxProfile need to be passed in alongside firefox-profile service. it was done but changed after some investigation to make it work....thank you 
before loading actual URL getting below errors
Firefox error :
![image](https://user-images.githubusercontent.com/17290705/154848858-2acfa14f-a34f-4ee9-b095-ed87f8127470.png)
Chrome:
 
![image](https://user-images.githubusercontent.com/17290705/154848927-7ad99a1c-7483-4037-ae39-43c6504bd30e.png)

observed newly added : 
extensions: [chromeModheader.getEncodedExtension()]
extensions: [firefoxModheader.getExtension()]

**Comment by RollerIn** [comment]
one more doubt , will this work for EDGE?...knew did not support safari
suppose if want to run on selenium grid ..will this library work?

**Comment by RollerIn** [comment]
Really your efforts are amazing.

**Comment by hao1300** [comment]
> Yes, Essentially, the firefoxProfile need to be passed in alongside firefox-profile service. it was done but changed after some investigation to make it work....thank you before loading actual URL getting below errors Firefox error : ![image](https://user-images.githubusercontent.com/17290705/154848858-2acfa14f-a34f-4ee9-b095-ed87f8127470.png) Chrome:
> 
> ![image](https://user-images.githubusercontent.com/17290705/154848927-7ad99a1c-7483-4037-ae39-43c6504bd30e.png)
> 
> observed newly added : extensions: [chromeModheader.getEncodedExtension()] extensions: [firefoxModheader.getExtension()]
> 
> and loading browsers also too slow

These are not errors. If you see the document title showing "Done" then it should actually be working.

I am not sure if this will work Edge in Selenium. I have not tried it. The ModHeader extension works for Edge, but I am not sure if there are any gotcha when using in Selenium WebDriver.
